src/domain/lightning-service.types.d.ts

```diff
@@ -0,0 +1,20 @@
+type LightningServiceErrorType =
+  | "UnknownLightningServiceError"
+  | "CouldNotDecodeReturnedPaymentRequest"
+type LightningServiceError = ErrorWithMessage<LightningServiceErrorType>
+
+type RegisterInvoiceArgs = {
+  description: string
+  satoshis: Satoshis
+  expiresAt: InvoiceExpiration
+}
+
+type RegisteredInvoice = {
+  invoice: LnInvoice
+}
+
+interface ILightningService {
+  registerInvoice(
+    registerInvoiceArgs: RegisterInvoiceArgs,
+  ): ResultAsync<RegisteredInvoice, LightningServiceError>
```

Comment:
Making a note to potentially add this line in a declaration file somewhere so that instances like these have access to `ResultAsync`

```ts
type ResultAsync<T, E> = import("neverthrow").ResultAsync<T, E>
```
---
![image](https://user-images.githubusercontent.com/17693119/128604598-43bb090f-3176-4f47-9c2c-d0bac58e3ddc.png)